### DISCUSSION TITLE
Attributes in form

### DISCUSSION THREAD

**Discussion by HeRm4nV** [reflex-dev/reflex#2071] [Q&A]
Hi everyone,

I want to add attributes to a form, something like this in html:

```<form class="css-0" name="form1" random_attribute>```

But ussing reflex i didn't find the answer to this, for now i have this code:

```
def multichoice_question(question: str, alternatives: list[str]) -> rx.Component:
    # Window
    return rx.vstack(
        # Question
        rx.form(
            rx.container(
                rx.text(
                    question,
                    padding_y= "5%",
                    font_size="1.5em",
                    color= Color.QUESTION_TEXT_COLOR.value,
                ),
                button_multichoice_alternatives(alternatives),
                text_align="center",
                margin_y="35%",
            ),
            on_submit=FormState.handle_submit({"question": question}),
        ),
        name_="form1",
        bg=Color.QUESTION_BACKGROUND.value,
        height= "100vh",
        width= "100vw",
    )

        

def button_multichoice_alternatives(alternatives: list[str]) -> rx.Component:
    return rx.center(
        rx.stack(
            rx.foreach(
                alternatives,
                alternative_button
            ),
            direction="column",
            align_items="center"
        )
    )

def alternative_button(text: str) -> rx.Component:
    return rx.button(
        text,
        #value=text,
        on_click= FormState.handle_alternative_button(text),
        type_="submit",
        # css
        border_radius= "2xl",
        width= "250%",
        font_size= "1.25em",
        bg= Color.BUTTON_QUESTION_BACKGROUND.value,
        color= Color.BUTTON_TEXT_COLOR.value,
        border= "2px solid #FFF",
    )
```

As you can see i can create forms without problems, with this i create random numbers of alternative buttons depending of the question, but the problem is this:

When i run the server i just obtained a form class, nothing about name or others attributes that i wanna add, something like "```<form class="css-0">```".

Do any of you knows any solution to this problem?

**Comment by masenf** [comment]
it looks like you're adding the `name_` prop on the `vstack`, not on the `form` component.

at any rate, only known props can be added to components, otherwise, they will be treated as style props and go into the css.

if you want to add arbitrary props to a component (and you know what you're doing), then you can pass a dict of `custom_attrs`, and whatever props are in there will be added to the component directly.
